src/app/api/generate-character/route.ts

```diff
@@ -0,0 +1,13 @@
+import { CharacterSheet } from '@/types';
+import { NextResponse } from 'next/server';
+
+export async function GET(request: Request) {
+  const { searchParams } = new URL(request.url);
+  const characterSheet: CharacterSheet = JSON.parse(decodeURIComponent(searchParams.get('characterSheet') || '{}'));
+
+  /* ... */
+
+  return {
```

Comment:
This isn't how you return from an API in Next! Check out [this page ](https://nextjs.org/docs/app/building-your-application/routing/router-handlers)to see how data is returned from this sort of call